### PR TITLE
Add CredentialRemoveRefrences API

### PR DESCRIPTION
**Pull request by talakad**
Signed-off-by: veda <veda@portworx.com>

**What this PR does / why we need it**: 
Allows users to delete any pending references to a credential 
**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/server/sdk/credentials_test.go [hunk 1]
```diff
@@ -1099,3 +1099,80 @@ func TestSdkCredentialGetOwnershipFromCred(t *testing.T) {
...
+	}
+
+	s.MockDriver().
+		EXPECT().
+		CredsEnumerate().
+		Return(enumerateData, nil)
+	s.MockDriver().
+		EXPECT().
+		CredsRemoveReferences(uuid).
+		Return(fmt.Errorf("Failed to remove refs"))
```
Comment (vjs-px) [inline api/server/sdk/credentials_test.go:1167]: I did not clearly get the difference between the TestSdkCredentialRemovereferencesSuccess() and TestSdkCredentialRemoveReferencesFailed()

Only the uuid is different and the mock driver return code is different. If the mock driver is not going to consult the enumerateData, then why even populate it?

Comment (talakad) [reply api/server/sdk/credentials_test.go:1167]: The goal is to test error response for the bad case.  We dont need to populate the data as such. but does not hurt.

File: api/server/sdk/credentials.go [hunk 2]
```diff
@@ -231,6 +231,41 @@ func (s *CredentialServer) Validate(
...
+func (s *CredentialServer) RemoveReferences(
+	ctx context.Context,
+	req *api.SdkCredentialRemoveReferencesRequest,
+) (*api.SdkCredentialRemoveReferencesResponse, error) {
+	if s.driver(ctx) == nil {
+		return nil, status.Error(codes.Unavailable, "Resource has not been initialized")
+	}
+
+	if len(req.GetCredentialId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must provide credentials uuid")
```
Comment (disrani-px) [inline api/server/sdk/credentials.go:244]: This can be name or uuid

File: api/api.proto [hunk 3]
```diff
@@ -2217,6 +2217,14 @@ service OpenStorageCredentials {
         get: "/v1/credentials/validate/{credential_id}"
       };
     }
+  // RemoveReferences is used to remove references to credentials
+  rpc RemoveReferences(SdkCredentialRemoveReferencesRequest)
```
Comment (disrani-px) [inline api/api.proto:2221]: I would structure this as DeleteReferences and add an API for GetReferences

The REST API paths could be DELETE /v1/credentials/references/{cred_id} and GET /v1/credentials/references/{cred_id}

File: api/server/sdk/credentials.go (hunk 2 above)
Comment (talakad) [reply api/server/sdk/credentials.go]: Yes. fixed the error.

File: api/api.proto (hunk 3 above)
Comment (talakad) [reply api/api.proto]: Done

**Pull request merged**